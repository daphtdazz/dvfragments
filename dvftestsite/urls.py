from django.urls import path

from dvftestapp.views import View1

urlpatterns = [
    path('', View1.as_view(), name='view1'),
]
