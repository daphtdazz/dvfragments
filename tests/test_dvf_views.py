from django.urls import reverse


def test_dvf_view(client, snapshot):
    r = client.get(reverse('view1'))

    assert r.status_code == 200
    assert snapshot == r.content.decode()

    r = client.get(reverse('view1') + '?fragment=if-frag')
    assert r.status_code == 200
    assert snapshot == r.content.decode()

    # this causes the template context to be different, so we get a different fragment
    r = client.get(reverse('view1') + '?fragment=if-frag&condition=true')
    assert r.status_code == 200
    assert snapshot == r.content.decode()


def test_dvf_view_initial(client, snapshot):
    # just check that if we ask for the fragment first that still works
    r = client.get(reverse('view1') + '?fragment=if-frag')
    assert r.status_code == 200
    assert snapshot == r.content.decode()
