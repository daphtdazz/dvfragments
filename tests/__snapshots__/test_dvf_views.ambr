# name: test_dvf_view
  '
  
  <!doctype html>
  <html>
      <head>
          <title>View 1</title>
          <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0">
      </head>
      <body>
          <h1>Hello world from view 1</h1>
          
          
          <p x-dvfragment-id=if-frag >Condition is false</p>
          
          125
          
      </body>
  </html>
  
  '
---
# name: test_dvf_view.1
  '
  
          
          <p x-dvfragment-id=if-frag >Condition is false</p>
          
          125
          
  '
---
# name: test_dvf_view.2
  '
  
          
          <p x-dvfragment-id=if-frag >Condition is true</p>
          
          125
          
  '
---
# name: test_dvf_view_initial
  '
  
          
          <p x-dvfragment-id=if-frag >Condition is false</p>
          
          125
          
  '
---
